### ISSUE TITLE
About loss of different modalities

### ISSUE THREAD

**Issue by PeiwenSun2000** [GeWu-Lab/OGM-GE_CVPR2022#36]
It has confused me for a long time that you have calculated the loss of different modalities but without backpropagation.

```
        loss = criterion(out, label)
        loss_v = criterion(out_v, label)
        loss_a = criterion(out_a, label)
        loss.backward()
```

Have you ever tried to implement backpropagation for the other 2 losses?

```
        loss = criterion(out, label)
        loss_v = criterion(out_v, label)
        loss_a = criterion(out_a, label)
        loss+=(loss_v+loss_a)
        loss.backward()
```
As far as I am concerned, the operation mentioned above would clearly help the network.
Would you happen to have some insights about this?

**Comment by echo0409** [comment]
Hi, thanks for your attention to our work.

In our method, we take the widely-used concatenation framework (concat uni-modal feature, and then fed into a classifier). loss_a and loss_v in the code are just for some obseservation and they could be removed for simplicity.

Your are right that adding additonal uni-modal loss into training could improve model performance, but it will also introduce additional two uni-modal classifier, bringing more training cost. Hence, we did not consider this training design.
